src/wallet/wallet.cpp

```diff
@@ -4945,23 +4945,20 @@ bool CWallet::Verify(interfaces::Chain& chain, const WalletLocation& location, s
     }
 
     // Keep same database environment instance across Verify/Recover calls below.
-    // Let tempWallet hold the pointer to the corresponding wallet database.
-    std::unique_ptr<CWallet> tempWallet = MakeUnique<CWallet>(chain, location, CreateWalletDatabase(wallet_path));
+    std::unique_ptr<WalletDatabase> database = CreateWalletDatabase(wallet_path);
```

Comment:
why change this?

Reply:
The presence of the `WalletDatabase` pointer was introduced in [bitcoin#11911](https://github.com/bitcoin/bitcoin/pull/11911/files#diff-1f2db0e4d5c12d109c7f0962333c245b49b696cb39ff432da048e9d6c08944d8R3880-R3881) though it wasn't being actively used back then.

For some reason, this was omitted during the backport process for that PR, which is problematic since it is invoked in [bitcoin#19324](https://github.com/bitcoin/bitcoin/pull/19324/commits/8f1bcf8b7b6e47c05f2e43dd98ec3505b888d8b3#diff-1f2db0e4d5c12d109c7f0962333c245b49b696cb39ff432da048e9d6c08944d8R3711), which this PR backports.